### PR TITLE
SWC-2259

### PR DESCRIPTION
**Pull request by Njanderson**
Remove unnecessary forms, and add enter behavior in form inputs, and fix multiple file uploads for Team icon uploading.

### CODE REVIEW

File: src/main/resources/org/sagebionetworks/web/client/widget/entity/editor/APITableColumnManagerViewImpl.ui.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 		<b:Modal ui:field="newColumnModal" title="Column Configuration"
 			fade="true" dataBackdrop="STATIC" dataKeyboard="false">
 			<b:ModalBody>
-				<b:Form type="HORIZONTAL">
```
Comment (jay-hodgson) [inline src/main/resources/org/sagebionetworks/web/client/widget/entity/editor/APITableColumnManagerViewImpl.ui.xml:17]: Could be a bh:Div, but in any case need addStyleNames="form-horizontal"

File: src/main/java/org/sagebionetworks/web/client/widget/entity/editor/ImageConfigViewImpl.java [hunk 2]
```diff
@@ -103,6 +104,17 @@ public void onClick(ClickEvent event) {
 			}
 		});
 		
+		ClickHandler upload = new ClickHandler() {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/editor/ImageConfigViewImpl.java]: remove

File: src/main/java/org/sagebionetworks/web/client/widget/entity/editor/ProvenanceConfigViewImpl.java [hunk 3]
```diff
@@ -43,6 +44,14 @@ public ProvenanceConfigViewImpl(ProvenanceConfigViewImplUiBinder binder, EntityF
 	}
 	
 	private void initClickHandlers() {
+		entityListField.addClickHandler(new ClickHandler() {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/editor/ProvenanceConfigViewImpl.java]: remove

File: src/main/java/org/sagebionetworks/web/client/view/NewAccountViewImpl.java [hunk 4]
```diff
@@ -89,7 +90,23 @@ public NewAccountViewImpl(NewAccountViewImplUiBinder binder,
 		init();
 	}
 	
+	// Apply to all input fields if clickEvent is enter
 	public void init() {
+		ClickHandler register = new ClickHandler() {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/view/NewAccountViewImpl.java]: Does this work?  I usually see a KeyDownHandler used for this purpose.

File: src/main/java/org/sagebionetworks/web/client/widget/entity/EditRegisteredTeamDialogViewImpl.java [hunk 5]
```diff
@@ -27,7 +30,14 @@
 	@Inject
 	public EditRegisteredTeamDialogViewImpl(RegisterTeamDialogViewImplUiBinder binder) {
 		modal = (Modal)binder.createAndBindUi(this);
```
Comment (Njanderson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/EditRegisteredTeamDialogViewImpl.java:32]: Where is this? I should be able to test this...

**Pull request merged**